lib/Dialect/AIE/Transforms/AIEObjectFifoStatefulTransform.cpp

```diff
@@ -217,12 +217,50 @@ struct AIEObjectFifoStatefulTransformPass
         }
     }
 
-    // Only test for this objfifo belonging to a LinkOp if we are in the shared
-    // memory case; otherwise, we will return `true` in any case.
+    // Check if the operation uses shared memory for linking.
+    // If shared memory is available and the link operation is valid,
+    // mark the operation for splitting and determine if it is used in a link
+    // operation. If the link operation is a distribute or join operation, DMAs
+    // are required even if shared memory is available. If the link has
+    // different memref types, then shared memory cannot be used, and DMAs are
+    // needed.
```

Comment:
Maybe something more like this would be clearer: 

Check if the objectfifo operation can use shared memory for linking.  If the link operation is a distribute or a join operation, or if the link has different memref types, DMAs are required even if shared memory is available and the objectfifo should be split. Otherwise check if the via_shared_memory attribute of the objectfifo operation is set and apply it.